Source/Plugin.BLE/Windows/Device.cs

```diff
@@ -168,9 +165,25 @@ public async Task<bool> ConnectInternal(ConnectParameters connectParameters, Can
                 MaybeRequestPreferredConnectionParameters(NativeDevice, connectParameters);
                 var devId = BluetoothDeviceId.FromId(NativeDevice.DeviceId);
                 gattSession = await GattSession.FromDeviceIdAsync(devId);
-                gattSession.MaintainConnection = true;
                 gattSession.SessionStatusChanged += GattSession_SessionStatusChanged;
                 gattSession.MaxPduSizeChanged += GattSession_MaxPduSizeChanged;
+                if (connectParameters.AutoPair)
+                {
+                    DeviceInformation deviceInformation = await DeviceInformation.CreateFromIdAsync(NativeDevice.DeviceId);
+                    if (!deviceInformation.Pairing.IsPaired && deviceInformation.Pairing.CanPair)
+                    {
+                        Trace.Message($"Pairing with {deviceInformation.Name} ({deviceInformation.Id})");
+                        deviceInformation.Pairing.Custom.PairingRequested += Custom_PairingRequested;
+                        DevicePairingResult result = await deviceInformation.Pairing.Custom.PairAsync(
+                            DevicePairingKinds.ConfirmOnly, DevicePairingProtectionLevel.Encryption);
+                        Trace.Message($"Pairing with {deviceInformation.Name} ({deviceInformation.Id}) result: {result.Status}");
+                        if (result.Status != DevicePairingResultStatus.Paired)
+                        {
+                            throw new Exception($"Could not pair: {result.Status}");
+                        }
```

Comment:
This part looks pretty similar to the implementation of `BondAsync` in #820, but with some small differences (e.g. the parameters of `PairAsync`). Are there supposed to be any differences? If not, one could put this into a common method that is used in both places (for consistency, and to avoid code duplication).
As mentioned, I think #820 should be merged first, so that this PR can be added on top of it.